In [4]:
# Importing essential libraries

# Data Gathering 
import pandas as pd
import numpy as np

# Data Visualization
import seaborn as sns



#### Data Gathering 

In [6]:
deliveries = pd.read_csv('deliveries.csv')
deliveries.sample(7)

match_id  inning                 batting_team           bowling_team  \
202301   1254094       1          Chennai Super Kings           Punjab Kings   
117294    829777       2             Delhi Daredevils       Rajasthan Royals   
246607   1422130       1          Sunrisers Hyderabad         Gujarat Titans   
182991   1216508       1               Mumbai Indians  Kolkata Knight Riders   
121306    829813       2  Royal Challengers Bangalore       Delhi Daredevils   
230605   1359494       1  Royal Challengers Bangalore         Delhi Capitals   
163133   1136614       1          Sunrisers Hyderabad  Kolkata Knight Riders   

        over  ball         batter             bowler   non_striker  \
202301     9     3       MS Dhoni       Ravi Bishnoi  F du Plessis   
117294    16     6      SS Tiwary         TG Southee     JP Duminy   
246607    10     6     AK Markram         Noor Ahmad     H Klaasen   
182991    11     4      SS Tiwary      Kuldeep Yadav     RG Sharma   
121306     1     1       CH Gayle             Z Khan       V Kohli   
230605     3     1        V Kohli           AR Patel  F du Plessis   
163133    17     3  CR Brathwaite  M Prasidh Krishna     MK Pandey   

        batsman_runs  extra_runs  total_runs extras_type  is_wicket  \
202301             0           0           0         NaN          0   
117294             1           0           1         NaN          0   
246607             1           0           1         NaN          0   
182991             0           0           0         NaN          0   
121306             0           0           0         NaN          0   
230605             1           0           1         NaN          0   
163133             1           0           1         NaN          0   

       player_dismissed dismissal_kind fielder  
202301              NaN            NaN     NaN  
117294              NaN            NaN     NaN  
246607              NaN            NaN     NaN  
182991              NaN            NaN     NaN  
121306              NaN            NaN     NaN  
230605              NaN            NaN     NaN  
163133              NaN            NaN     NaN

`Column` | Description
:-|:-
`match_id` |	Unique identifier for each match, used to link with match details.
`inning`|	Indicates the innings (1 or 2) in the match.
`batting_team`|Name of the team batting in the given inning.
`bowling_team`|	Name of the team bowling in the given inning.
`over`|	The over number in the current inning (e.g., 1, 2, ..., 20).
`ball`|	The ball number within the over (1 to 6).
`batter`|	Name of the player currently batting.
`bowler`	|Name of the player currently bowling.
`non_striker`|	Name of the non-striker at the other end of the pitch.
`batsman_runs`|	Number of runs scored by the batsman on the current delivery (can include 0 for dot balls or 4/6 for boundaries).
`extra_runs`	|Number of extra runs (such as no-balls, wides) on this delivery.
`total_runs`	|Total runs scored on this ball (sum of batsman runs and extra runs).
`extras_type`	|Type of extra run, if applicable (e.g., wide, no-ball, bye, legbye).
`is_wicket`	|Indicates if the delivery resulted in a wicket (1 for wicket, 0 for no wicket).
`player_dismissed`|	Name of the player dismissed on the ball (if applicable, otherwise null).
`dismissal_kind`	|Type of dismissal (e.g., bowled, caught, run out, LBW).
`fielder`	|Name of the fielder involved in the dismissal (for dismissals like catch, run out, etc.; null for dismissals not involving a fielder).

In [8]:
matches = pd.read_csv('matches.csv')
matches.sample(7)

id season       city        date         match_type  \
459    829707   2015    Chennai  2015-04-09             League   
287    548348   2012    Cuttack  2012-05-01             League   
948   1312199   2022  Ahmedabad  2022-05-27        Qualifier 2   
914   1304085   2022       Pune  2022-04-26             League   
825   1254067   2021    Chennai  2021-04-18             League   
1000  1359525   2023  Ahmedabad  2023-05-07             League   
245    501269   2011     Mumbai  2011-05-25  Elimination Final   

     player_of_match                                          venue  \
459          A Nehra                MA Chidambaram Stadium, Chepauk   
287    KC Sangakkara                               Barabati Stadium   
948       JC Buttler               Narendra Modi Stadium, Ahmedabad   
914          R Parag  Maharashtra Cricket Association Stadium, Pune   
825   AB de Villiers       MA Chidambaram Stadium, Chepauk, Chennai   
1000    Shubman Gill               Narendra Modi Stadium, Ahmedabad   
245         MM Patel                               Wankhede Stadium   

                            team1                        team2  \
459           Chennai Super Kings             Delhi Daredevils   
287               Deccan Chargers                Pune Warriors   
948   Royal Challengers Bangalore             Rajasthan Royals   
914              Rajasthan Royals  Royal Challengers Bangalore   
825   Royal Challengers Bangalore        Kolkata Knight Riders   
1000               Gujarat Titans         Lucknow Super Giants   
245                Mumbai Indians        Kolkata Knight Riders   

                      toss_winner toss_decision                       winner  \
459              Delhi Daredevils         field          Chennai Super Kings   
287               Deccan Chargers           bat              Deccan Chargers   
948              Rajasthan Royals         field             Rajasthan Royals   
914   Royal Challengers Bangalore         field             Rajasthan Royals   
825   Royal Challengers Bangalore           bat  Royal Challengers Bangalore   
1000         Lucknow Super Giants         field               Gujarat Titans   
245                Mumbai Indians         field               Mumbai Indians   

       result  result_margin  target_runs  target_overs super_over method  \
459      runs            1.0        151.0          20.0          N    NaN   
287      runs           13.0        187.0          20.0          N    NaN   
948   wickets            7.0        158.0          20.0          N    NaN   
914      runs           29.0        145.0          20.0          N    NaN   
825      runs           38.0        205.0          20.0          N    NaN   
1000     runs           56.0        228.0          20.0          N    NaN   
245   wickets            4.0        148.0          20.0          N    NaN   

             umpire1                umpire2  
459   RK Illingworth            VA Kulkarni  
287        Aleem Dar           AK Chaudhary  
948      CB Gaffaney            Nitin Menon  
914     BNJ Oxenford  KN Ananthapadmanabhan  
825    C Shamshuddin            Nitin Menon  
1000    AK Chaudhary         A Nand Kishore  
245        Asad Rauf             SJA Taufel

`Column`	|Description
:-|:-
`id`	|Unique identifier for each match, similar to match_id in the ball-by-ball dataset, useful for linking datasets.
`season`|	Year in which the IPL season took place.
`city`	|City where the match was played.
`date`	|Date of the match.
`match_type`|	Type of match (could indicate regular season, playoff, or final).
`player_of_match`|	Name of the player awarded "Player of the Match."
`venue`|	Stadium where the match was played.
`team1`	|Name of the first team (order doesn’t imply batting or bowling first).
`team2`	|Name of the second team.
`toss_winner`|	Team that won the toss.
`toss_decision`|	Decision taken by the toss-winning team (bat or field).
`winner`|Name of the team that won the match.
`result`|	Outcome type of the match (e.g., normal, tie, no result).
`result_margin`|	Margin by which the team won (e.g., runs or wickets).
`target_runs`	|Target runs set for the chasing team (null if not applicable).
`target_overs`	|Target overs if the match is shortened (e.g., in rain-affected matches; null if not applicable).
`super_over`	|Indicates if the match had a super over (1 for yes, 0 for no).
`method`	|Method used to determine the result in rain-affected games (e.g., Duckworth-Lewis-Stern, or null if not applicable).
`umpire1`	|Name of the first on-field umpire.
`umpire2`	|Name of the second on-field umpire.

- here we have two datasets one containing ball by ball details and other containing match wise details.
- we are going to analyse each data set for insights.


#### Data Cleaning

In [12]:
deliveries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 260920 entries, 0 to 260919
Data columns (total 17 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   match_id          260920 non-null  int64 
 1   inning            260920 non-null  int64 
 2   batting_team      260920 non-null  object
 3   bowling_team      260920 non-null  object
 4   over              260920 non-null  int64 
 5   ball              260920 non-null  int64 
 6   batter            260920 non-null  object
 7   bowler            260920 non-null  object
 8   non_striker       260920 non-null  object
 9   batsman_runs      260920 non-null  int64 
 10  extra_runs        260920 non-null  int64 
 11  total_runs        260920 non-null  int64 
 12  extras_type       14125 non-null   object
 13  is_wicket         260920 non-null  int64 
 14  player_dismissed  12950 non-null   object
 15  dismissal_kind    12950 non-null   object
 16  fielder           9354 non-null    obj

- here we get data information where we know basics about our data
    - There are 260920 rows and 17 columns in deliveries dataset.
    - some columns have missing values.


In [14]:
matches.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1095 entries, 0 to 1094
Data columns (total 20 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   id               1095 non-null   int64  
 1   season           1095 non-null   object 
 2   city             1044 non-null   object 
 3   date             1095 non-null   object 
 4   match_type       1095 non-null   object 
 5   player_of_match  1090 non-null   object 
 6   venue            1095 non-null   object 
 7   team1            1095 non-null   object 
 8   team2            1095 non-null   object 
 9   toss_winner      1095 non-null   object 
 10  toss_decision    1095 non-null   object 
 11  winner           1090 non-null   object 
 12  result           1095 non-null   object 
 13  result_margin    1076 non-null   float64
 14  target_runs      1092 non-null   float64
 15  target_overs     1092 non-null   float64
 16  super_over       1095 non-null   object 
 17  method        

- Here  we have 1095 rows and 20 columns .
- there are some missing values in this data.
- Date column is in form of object which can be converted into date time for better analysis.

##### Further I cleaned both datas column wise 

#### Deliveries cleaning 

In [18]:
deliveries.sample(25)

match_id  inning                 batting_team  \
118981    829791       2        Kolkata Knight Riders   
231417   1359497       1               Gujarat Titans   
57834     501268       2          Chennai Super Kings   
175503   1178420       1          Sunrisers Hyderabad   
226198   1359476       1                 Punjab Kings   
28052     419109       2        Kolkata Knight Riders   
169522   1178395       2               Delhi Capitals   
227842   1359482       2             Rajasthan Royals   
86608     598040       1        Kolkata Knight Riders   
221654   1304103       1               Gujarat Titans   
216422   1304081       1               Gujarat Titans   
154435   1136577       2             Rajasthan Royals   
169005   1178393       1          Chennai Super Kings   
5694      336006       1  Royal Challengers Bangalore   
227740   1359482       1                 Punjab Kings   
27703     419108       1              Kings XI Punjab   
139473   1082603       1       Rising Pune Supergiant   
141049   1082609       2              Kings XI Punjab   
187467   1216526       2               Mumbai Indians   
47948     501225       2             Rajasthan Royals   
105632    734023       1          Chennai Super Kings   
108394    734047       1              Kings XI Punjab   
91996     598062       2             Delhi Daredevils   
42013     501199       2             Rajasthan Royals   
227498   1359481       1               Delhi Capitals   

                       bowling_team  over  ball           batter  \
118981              Kings XI Punjab    17     5        PP Chawla   
231417             Rajasthan Royals    15     6        DA Miller   
57834   Royal Challengers Bangalore    13     6         SK Raina   
175503             Rajasthan Royals    16     1  Shakib Al Hasan   
226198        Kolkata Knight Riders     0     2   P Simran Singh   
28052   Royal Challengers Bangalore    12     7       SC Ganguly   
169522  Royal Challengers Bangalore     0     7          SS Iyer   
227842                 Punjab Kings    10     2       D Padikkal   
86608              Delhi Daredevils    11     2           DB Das   
221654         Lucknow Super Giants     4     5     Shubman Gill   
216422        Kolkata Knight Riders    17     1        R Tewatia   
154435          Chennai Super Kings    11     6      RA Tripathi   
169005              Kings XI Punjab    15     3         MS Dhoni   
5694                Kings XI Punjab     0     7        JH Kallis   
227740             Rajasthan Royals    14     3        JM Sharma   
27703              Delhi Daredevils    16     1         RR Powar   
139473                Gujarat Lions     6     6        SPD Smith   
141049          Sunrisers Hyderabad     2     3       GJ Maxwell   
187467        Kolkata Knight Riders     5     2        RG Sharma   
47948          Kochi Tuskers Kerala    10     6          J Botha   
105632        Kolkata Knight Riders    13     3     F du Plessis   
108394          Chennai Super Kings     7     2         V Sehwag   
91996           Chennai Super Kings    17     2        IK Pathan   
42013               Deccan Chargers     8     6          J Botha   
227498               Gujarat Titans    13     6         AR Patel   

                bowler non_striker  batsman_runs  extra_runs  total_runs  \
118981    BE Hendricks     GB Hogg             4           0           4   
231417  Sandeep Sharma   A Manohar             1           0           1   
57834         CH Gayle    MS Dhoni             0           0           0   
175503      JD Unadkat    DJ Hooda             1           0           1   
226198        UT Yadav    S Dhawan             2           0           2   
28052         DW Steyn    BJ Hodge             2           0           2   
169522      TG Southee     PP Shaw             0           1           1   
227842        NT Ellis   SV Samson             0           0           0   
86608         S Nadeem    R Bhatia             0           0           0

#### match_id

In [20]:
deliveries['match_id'].isnull().sum()

0

In [21]:
deliveries['match_id'].nunique()

1095

- In match_id column there isn't any null value .
- Their are 1095 unique match id's which matches #rows in matches dataset.
   i.e. We have information of all matches.

#### inning

In [24]:
deliveries['inning'].isnull().sum()

0

In [25]:
deliveries['inning'].unique()

array([1, 2, 3, 4, 5, 6], dtype=int64)

- here data does not have null values.
- but innings values are little disturbed, as we know their should be only 2 innings but have till 6.
    - 3rd and 4th inning is their due to a super over but 5th and 6th inning is a entry error.

In [27]:
# If inning are 5 changing to 3 and if 6 changed to 4
def inning_replacing(inning):
    if inning == 5:
        return 3
    elif inning == 6:
        return 4
    else:
        return inning

In [28]:
deliveries['inning']=deliveries['inning'].apply(inning_replacing)

In [29]:
deliveries['inning'].unique()

array([1, 2, 3, 4], dtype=int64)

#### batting team

In [31]:
deliveries['batting_team'].isnull().sum()

0

In [32]:
deliveries['batting_team'].nunique()

19

In [33]:
deliveries['batting_team'].unique()

array(['Kolkata Knight Riders', 'Royal Challengers Bangalore',
       'Chennai Super Kings', 'Kings XI Punjab', 'Rajasthan Royals',
       'Delhi Daredevils', 'Mumbai Indians', 'Deccan Chargers',
       'Kochi Tuskers Kerala', 'Pune Warriors', 'Sunrisers Hyderabad',
       'Rising Pune Supergiants', 'Gujarat Lions',
       'Rising Pune Supergiant', 'Delhi Capitals', 'Punjab Kings',
       'Lucknow Super Giants', 'Gujarat Titans',
       'Royal Challengers Bengaluru'], dtype=object)

- their is no null value in batting team
- over the years names of teams are changed so same team is present with several names or their are some speeling difference

In [35]:
# correcting teams name if differs
def change_team_name(team_name):
    if team_name in ['Rising Pune Supergiants','Pune Warriors']:
        return 'Rising Pune Supergiant'
    elif team_name == 'Royal Challengers Bengaluru':
        return 'Royal Challengers Bangalore'
    elif team_name == 'Deccan Chargers':
        return 'Sunrisers Hyderabad'
    elif team_name == 'Delhi Daredevils':
        return 'Delhi Capitals'
    elif team_name == 'Gujarat Lions':
        return 'Gujarat Titans'
    elif team_name == 'Kings XI Punjab':
        return 'Punjab Kings'
    else:
        return team_name

In [36]:
deliveries['batting_team'] = deliveries['batting_team'].apply(change_team_name)

In [37]:
deliveries['batting_team'].nunique()

12

- there were 19 teams in batting team column which reduced to 12 as we know till today 12 teams were played in IPL

#### bowling team

In [40]:
deliveries['bowling_team'].isnull().sum()

0

In [41]:
deliveries['bowling_team'].nunique()

19

In [42]:
deliveries['bowling_team'] = deliveries['bowling_team'].apply(change_team_name)

In [43]:
deliveries['bowling_team'].nunique()

12

- their were 19 teams reduced to 12 with same problem as batting team

#### Over

In [46]:
deliveries['over'].isnull().sum()

0

In [47]:
deliveries['over'].unique()

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19], dtype=int64)

- 20 overs are present 1 over is calculated after completing.

#### ball

In [50]:
deliveries['ball'].isnull().sum()

0

In [51]:
deliveries['ball'].unique()

array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11], dtype=int64)

- as we know each over has 6 balls their can be wides and no balls in some overs.

#### batter

In [54]:
deliveries['batter'].isnull().sum()

0

In [55]:
deliveries['batter'].nunique()

673

In [56]:
deliveries['batter'].unique()

array(['SC Ganguly', 'BB McCullum', 'RT Ponting', 'DJ Hussey',
       'Mohammad Hafeez', 'R Dravid', 'W Jaffer', 'V Kohli', 'JH Kallis',
       'CL White', 'MV Boucher', 'B Akhil', 'AA Noffke', 'P Kumar',
       'Z Khan', 'SB Joshi', 'PA Patel', 'ML Hayden', 'MEK Hussey',
       'MS Dhoni', 'SK Raina', 'JDP Oram', 'S Badrinath', 'K Goel',
       'JR Hopes', 'KC Sangakkara', 'Yuvraj Singh', 'SM Katich',
       'IK Pathan', 'T Kohli', 'YK Pathan', 'SR Watson', 'M Kaif',
       'DS Lehmann', 'RA Jadeja', 'M Rawat', 'D Salunkhe', 'SK Warne',
       'SK Trivedi', 'G Gambhir', 'V Sehwag', 'S Dhawan', 'L Ronchi',
       'ST Jayasuriya', 'DJ Thornely', 'RV Uthappa', 'PR Shah',
       'AM Nayar', 'SM Pollock', 'Harbhajan Singh', 'S Chanderpaul',
       'LRPL Taylor', 'AC Gilchrist', 'Y Venugopal Rao', 'VVS Laxman',
       'A Symonds', 'RG Sharma', 'SB Styris', 'AS Yadav', 'SB Bangar',
       'WPUJC Vaas', 'RP Singh', 'WP Saha', 'LR Shukla',
       'DPMD Jayawardene', 'S Sohal', 'B Lee', 'PP Cha

- there are total 673 batter played in IPL till now

#### bowler

In [59]:
deliveries['bowler'].isnull().sum()

0

In [60]:
deliveries['bowler'].nunique()

530

In [61]:
deliveries['bowler'].unique()

array(['P Kumar', 'Z Khan', 'AA Noffke', 'JH Kallis', 'SB Joshi',
       'CL White', 'AB Dinda', 'I Sharma', 'AB Agarkar', 'SC Ganguly',
       'LR Shukla', 'B Lee', 'S Sreesanth', 'JR Hopes', 'IK Pathan',
       'K Goel', 'PP Chawla', 'WA Mota', 'JDP Oram', 'MS Gony',
       'M Muralitharan', 'P Amarnath', 'Joginder Sharma', 'GD McGrath',
       'B Geeves', 'MF Maharoof', 'R Bhatia', 'DL Vettori', 'MM Patel',
       'SR Watson', 'SK Trivedi', 'SK Warne', 'YK Pathan', 'D Salunkhe',
       'R Vinay Kumar', 'B Akhil', 'A Nehra', 'SM Pollock', 'DS Kulkarni',
       'ST Jayasuriya', 'Harbhajan Singh', 'AM Nayar', 'M Kartik',
       'Mohammad Hafeez', 'DJ Hussey', 'WPUJC Vaas', 'RP Singh',
       'SB Styris', 'SB Bangar', 'A Symonds', 'PP Ojha', 'Pankaj Singh',
       'Mohammad Asif', 'VY Mahesh', 'Shahid Afridi', 'DJ Bravo',
       'VS Yeligati', 'MA Khote', 'D Kalyankrishna', 'VRV Singh',
       'Sohail Tanvir', 'A Kumble', 'DNT Zoysa', 'SD Chitnis',
       'Yuvraj Singh', 'Shoaib Malik',

- 530 bowlers played IPL till now

#### non-striker

In [64]:
deliveries['non_striker'].isnull().sum()

0

In [65]:
deliveries['non_striker'].nunique()

663

In [66]:
deliveries['non_striker'].unique()

array(['BB McCullum', 'SC Ganguly', 'RT Ponting', 'DJ Hussey',
       'Mohammad Hafeez', 'W Jaffer', 'R Dravid', 'V Kohli', 'JH Kallis',
       'CL White', 'MV Boucher', 'B Akhil', 'AA Noffke', 'P Kumar',
       'Z Khan', 'SB Joshi', 'ML Hayden', 'PA Patel', 'MEK Hussey',
       'MS Dhoni', 'SK Raina', 'JDP Oram', 'S Badrinath', 'JR Hopes',
       'K Goel', 'KC Sangakkara', 'Yuvraj Singh', 'SM Katich',
       'IK Pathan', 'YK Pathan', 'T Kohli', 'SR Watson', 'M Kaif',
       'DS Lehmann', 'RA Jadeja', 'M Rawat', 'D Salunkhe', 'SK Warne',
       'SK Trivedi', 'V Sehwag', 'G Gambhir', 'S Dhawan', 'ST Jayasuriya',
       'L Ronchi', 'DJ Thornely', 'RV Uthappa', 'PR Shah', 'AM Nayar',
       'SM Pollock', 'Harbhajan Singh', 'S Chanderpaul', 'LRPL Taylor',
       'Y Venugopal Rao', 'AC Gilchrist', 'VVS Laxman', 'A Symonds',
       'SB Styris', 'AS Yadav', 'SB Bangar', 'WPUJC Vaas', 'RP Singh',
       'PP Ojha', 'WP Saha', 'LR Shukla', 'DPMD Jayawardene', 'S Sohal',
       'PP Chawla', 'WA M

- there are total 663 unique values in non-striker

#### batsmam_runs

In [69]:
deliveries['batsman_runs'].isnull().sum()

0

In [70]:
deliveries['batsman_runs'].unique()

array([0, 4, 6, 1, 2, 5, 3], dtype=int64)

- values are till 6 that a batter can strike

#### extra_runs

In [73]:
deliveries['extra_runs'].isnull().sum()

0

In [74]:
deliveries['extra_runs'].unique()

array([1, 0, 5, 4, 2, 3, 7], dtype=int64)

- because of bowlers mistake some penalties are gives so the value is 7

heelo i think i am loosing my ind

In [77]:
deliveries

match_id  inning           batting_team                 bowling_team  \
0         335982       1  Kolkata Knight Riders  Royal Challengers Bangalore   
1         335982       1  Kolkata Knight Riders  Royal Challengers Bangalore   
2         335982       1  Kolkata Knight Riders  Royal Challengers Bangalore   
3         335982       1  Kolkata Knight Riders  Royal Challengers Bangalore   
4         335982       1  Kolkata Knight Riders  Royal Challengers Bangalore   
...          ...     ...                    ...                          ...   
260915   1426312       2  Kolkata Knight Riders          Sunrisers Hyderabad   
260916   1426312       2  Kolkata Knight Riders          Sunrisers Hyderabad   
260917   1426312       2  Kolkata Knight Riders          Sunrisers Hyderabad   
260918   1426312       2  Kolkata Knight Riders          Sunrisers Hyderabad   
260919   1426312       2  Kolkata Knight Riders          Sunrisers Hyderabad   

        over  ball       batter         bowler  non_striker  batsman_runs  \
0          0     1   SC Ganguly        P Kumar  BB McCullum             0   
1          0     2  BB McCullum        P Kumar   SC Ganguly             0   
2          0     3  BB McCullum        P Kumar   SC Ganguly             0   
3          0     4  BB McCullum        P Kumar   SC Ganguly             0   
4          0     5  BB McCullum        P Kumar   SC Ganguly             0   
...      ...   ...          ...            ...          ...           ...   
260915     9     5      SS Iyer     AK Markram      VR Iyer             1   
260916     9     6      VR Iyer     AK Markram      SS Iyer             1   
260917    10     1      VR Iyer  Shahbaz Ahmed      SS Iyer             1   
260918    10     2      SS Iyer  Shahbaz Ahmed      VR Iyer             1   
260919    10     3      VR Iyer  Shahbaz Ahmed      SS Iyer             1   

        extra_runs  total_runs extras_type  is_wicket player_dismissed  \
0                1           1     legbyes          0              NaN   
1                0           0         NaN          0              NaN   
2                1           1       wides          0              NaN   
3                0           0         NaN          0              NaN   
4                0           0         NaN          0              NaN   
...            ...         ...         ...        ...              ...   
260915           0           1         NaN          0              NaN   
260916           0           1         NaN          0              NaN   
260917           0           1         NaN          0              NaN   
260918           0           1         NaN          0              NaN   
260919           0           1         NaN          0              NaN   

       dismissal_kind fielder  
0                 NaN     NaN  
1                 NaN     NaN  
2                 NaN     NaN  
3                 NaN     NaN  
4                 NaN     NaN  
...               ...     ...  
260915            NaN     NaN  
260916            NaN     NaN  
260917            NaN     NaN  
260918            NaN     NaN  
260919            NaN     NaN  

[260920 rows x 17 columns]